tests/backfill/test_parse_backfill.py

```diff
@@ -0,0 +1,358 @@
+        with pytest.raises(ValueError) as e:
+            invalid_excluded_dates = [TEST_BACKFILL_1.end_date + timedelta(days=1)]
+            Backfill(
+                TEST_BACKFILL_1.entry_date,
+                TEST_BACKFILL_1.start_date,
+                TEST_BACKFILL_1.end_date,
+                invalid_excluded_dates,
+                TEST_BACKFILL_1.reason,
+                TEST_BACKFILL_1.watchers,
+                TEST_BACKFILL_1.status,
+            )
+
+        assert "Invalid excluded dates" in str(e.value)
+
+    def test_invalid_excluded_dates_greater_than_end_date_multiple(self):
+        with pytest.raises(ValueError) as e:
+            invalid_excluded_dates = [
+                TEST_BACKFILL_1.end_date,
+                TEST_BACKFILL_1.end_date + timedelta(days=1),
+            ]
+            Backfill(
+                TEST_BACKFILL_1.entry_date,
+                TEST_BACKFILL_1.start_date,
+                TEST_BACKFILL_1.end_date,
+                invalid_excluded_dates,
+                TEST_BACKFILL_1.reason,
+                TEST_BACKFILL_1.watchers,
+                TEST_BACKFILL_1.status,
+            )
+
+        assert "Invalid excluded dates" in str(e.value)
+
+    def test_invalid_excluded_dates_less_than_start_date(self):
+        with pytest.raises(ValueError) as e:
+            invalid_excluded_dates = [TEST_BACKFILL_1.start_date - timedelta(days=1)]
+            Backfill(
+                TEST_BACKFILL_1.entry_date,
+                TEST_BACKFILL_1.start_date,
+                TEST_BACKFILL_1.end_date,
+                invalid_excluded_dates,
+                TEST_BACKFILL_1.reason,
+                TEST_BACKFILL_1.watchers,
+                TEST_BACKFILL_1.status,
+            )
+
+        assert "Invalid excluded dates" in str(e.value)
+
+    def test_invalid_excluded_dates_less_than_start_date_multiple(self):
+        with pytest.raises(ValueError) as e:
+            invalid_excluded_dates = [
+                TEST_BACKFILL_1.start_date,
+                TEST_BACKFILL_1.start_date - timedelta(days=1),
+            ]
+            Backfill(
+                TEST_BACKFILL_1.entry_date,
+                TEST_BACKFILL_1.start_date,
+                TEST_BACKFILL_1.end_date,
+                invalid_excluded_dates,
+                TEST_BACKFILL_1.reason,
+                TEST_BACKFILL_1.watchers,
+                TEST_BACKFILL_1.status,
+            )
+
+        assert "Invalid excluded dates" in str(e.value)
+
+    def test_invalid_status(self):
+        with pytest.raises(AttributeError):
+            invalid_status = "invalid_status"
+            Backfill(
+                TEST_BACKFILL_1.entry_date,
+                TEST_BACKFILL_1.start_date,
+                TEST_BACKFILL_1.end_date,
+                TEST_BACKFILL_1.excluded_dates,
+                TEST_BACKFILL_1.reason,
+                TEST_BACKFILL_1.watchers,
+                invalid_status,
+            )
+
+    def test_non_existing_file(self):
```

Comment:
Since the expected output for the test and the test logic are the same we could avoid test logic duplication here by parametizing the tests:
https://towardsdatascience.com/how-to-parameterize-python-tests-using-pytest-e8800bf288c5

Reply:
I will take note to explore this at the next iteration